src/MongoDB/Driver/Manager.cpp

```diff
@@ -238,34 +238,34 @@ static bool hippo_mongo_driver_manager_apply_rp(mongoc_uri_t *uri, const Array o
 		mongoc_read_prefs_set_tags(new_rp, b_tags);
 	}
 
-	/* Handle maxStalenessMS, and make sure it is not combined with PRIMARY readPreference */
+	/* Handle maxStalenessSeconds, and make sure it is not combined with PRIMARY readPreference */
 	if (
-		(options.exists(s_MongoDBDriverManager_maxstalenessms) && options[s_MongoDBDriverManager_maxstalenessms].isInteger())
+		(options.exists(s_MongoDBDriverManager_maxstalenessseconds) && options[s_MongoDBDriverManager_maxstalenessseconds].isInteger())
 		||
-		(options.exists(s_MongoDBDriverManager_maxStalenessMS) && options[s_MongoDBDriverManager_maxStalenessMS].isInteger())
+		(options.exists(s_MongoDBDriverManager_maxStalenessSeconds) && options[s_MongoDBDriverManager_maxStalenessSeconds].isInteger())
 	) {
-		int64_t maxStalenessMS = 0;
+		int64_t maxStalenessSeconds = 0;
 
 		if (mongoc_read_prefs_get_mode(new_rp) == MONGOC_READ_PRIMARY) {
-			throw MongoDriver::Utils::throwInvalidArgumentException("Primary read preference mode conflicts with maxStalenessMS");
+			throw MongoDriver::Utils::throwInvalidArgumentException("Primary read preference mode conflicts with maxStalenessSeconds");
 			mongoc_read_prefs_destroy(new_rp);
 
 			return false;
 		}
 
-		if (options.exists(s_MongoDBDriverManager_maxstalenessms)) {
-			maxStalenessMS = options[s_MongoDBDriverManager_maxstalenessms].toInt64();
+		if (options.exists(s_MongoDBDriverManager_maxstalenessseconds)) {
+			maxStalenessSeconds = options[s_MongoDBDriverManager_maxstalenessseconds].toInt64();
 		} else {
-			maxStalenessMS = options[s_MongoDBDriverManager_maxStalenessMS].toInt64();
+			maxStalenessSeconds = options[s_MongoDBDriverManager_maxStalenessSeconds].toInt64();
 		}
 
-		if (maxStalenessMS < 0) {
-			throw MongoDriver::Utils::throwInvalidArgumentException("Expected maxStalenessMS to be >= 0, " + Variant(maxStalenessMS).toString() + " given");
+		if (maxStalenessSeconds < 0) {
+			throw MongoDriver::Utils::throwInvalidArgumentException("Expected maxStalenessSeconds to be >= 0, " + Variant(maxStalenessSeconds).toString() + " given");
```

Comment:
Related to my earlier comment for ReadPreference constructor validation, I think we should NOP on -1 and then throw if the value is <= 0. The exception should tell users that we require a value greater than 0.